### PR TITLE
ocrvs-2577: three dots menu made available only for national sysadmin, and sysadmin just for his office

### PR DESCRIPTION
**Pull request by emon-swe-sust**
Screenshot from 2022-02-17 12-14-14
Screenshot from 2022-02-17 12-14-37
Screenshot from 2022-02-17 12-15-14

### CODE REVIEW

File: packages/client/src/views/SysAdmin/Team/user/UserList.tsx [hunk 1]
```diff
@@ -466,25 +466,39 @@ function UserListComponent(props: IProps) {
...
 
   function getStatusMenuType(
+    userDetails: IUserDetails | null,
+    locationId: string,
     user: GQLUser,
     index: number,
     status?: string,
     underInvestigation?: boolean
   ) {
+    const showMenuIcon =
```
Comment (rikukissa) [inline packages/client/src/views/SysAdmin/Team/user/UserList.tsx:476]: I think you can make this a little simpler by doing

```suggestion
    const canEditUserDetails = 
      userDetails && ['NATIONAL_SYSTEM_ADMIN', 'LOCAL_SYSTEM_ADMIN'].includes(userDetails.role) &&
      userDetails.primaryOffice?.id === locationId
```

File: packages/client/src/views/SysAdmin/Team/user/UserList.tsx [hunk 2]
```diff
@@ -466,25 +466,39 @@ function UserListComponent(props: IProps) {
   }
 
   function getStatusMenuType(
+    userDetails: IUserDetails | null,
+    locationId: string,
     user: GQLUser,
```
Comment (rikukissa) [inline packages/client/src/views/SysAdmin/Team/user/UserList.tsx:471]: Could you get the user id from `userDetails`? Could this parameter be dropped altogether?

**Review by rikukissa — APPROVED** [review]

File: packages/client/src/views/SysAdmin/Team/user/UserList.tsx (hunk 1 above)
Comment (emon-swe-sust) [reply packages/client/src/views/SysAdmin/Team/user/UserList.tsx]: fixed it!

File: packages/client/src/views/SysAdmin/Team/user/UserList.tsx (hunk 2 above)
Comment (emon-swe-sust) [reply packages/client/src/views/SysAdmin/Team/user/UserList.tsx:471]: I thought userDetails is for logged in user's details, and user is the information for the users in the userlist rows. they are not same. so I don't think this parameter can be dropped.

**Pull request merged**